src/admin/helpers/MenuHelper.php

```diff
@@ -255,18 +256,62 @@ public static function getContainers()
     public static function getWebsites()
     {
         if (self::$websites === null) {
-            self::$websites = (new Query())
-                ->select(['cms_website.id', 'cms_website.name', 'cms_website.host', 'cms_website.is_default', 'default_container_id' => 'MIN(cms_nav_container.id)'])
+            $websites = (new Query())
+                ->select([
+                    'cms_website.id',
+                    'cms_website.name',
+                    'cms_website.host',
+                    'cms_website.is_default',
+                    'default_container_id' => 'MIN(cms_nav_container.id)',
+                    'group_ids',
+                    'user_ids'
+                ])
                 ->from('cms_website')
-                ->innerJoin('cms_nav_container', 'website_id = cms_website.id')
+                ->leftJoin('cms_nav_container', 'website_id = cms_website.id')
                 ->where(['cms_website.is_active' => true, 'cms_website.is_deleted' => false])
                 ->groupBy('cms_website.id')
                 ->all();
+
+            foreach ($websites as $websiteIndex => $website) {
+                if (!self::checkUserWebsitePermissions($website)) {
+                    unset($websites[$websiteIndex]);
+                }
+            }
+
+            self::$websites = array_values($websites);
         }
         
         return self::$websites;
     }
     
+    /**
+     * Check that the current user is allowed to access to the given website.
+     *
+     * @param array $website
```

Comment:
Could you please provide that this array must have an user_ids and group_ids key? even its a private function, we will stumble upon once again, i am sure about that :-)